napari/layers/base/base.py

```diff
@@ -431,14 +531,45 @@ def _extent_world(self) -> np.ndarray:
         -------
         extent_world : array, shape (2, D)
         """
-        return self._transforms['data2world'](self._extent_data)
+        # Get full nD bounding box
+        data_extent = self._extent_data
+        D = data_extent.shape[1]
+        full_data_extent = np.array(np.meshgrid(*data_extent.T)).T.reshape(
+            -1, D
+        )
+        full_world_extent = self._transforms['data2world'](full_data_extent)
+        world_extent = np.array(
+            [
+                np.min(full_world_extent, axis=0),
+                np.max(full_world_extent, axis=0),
+            ]
+        )
+        return world_extent
 
     @property
     def _slice_indices(self):
         """(D, ) array: Slice indices in data coordinates."""
-        world_pts = [self._dims_point[ax] for ax in self.dims.not_displayed]
+        # clipping plane in world coordinates
+        # clipping_plane = [1, 0, 0]
         inv_transform = self._transforms['data2world'].inverse
-        data_pts = inv_transform.set_slice(self.dims.not_displayed)(world_pts)
+        # data_clipping_plane = inv_transform(clipping_plane)
+
+        if self.ndim > self.dims.ndisplay:
+            clipping_plane = np.ones(self.ndim)
+            clipping_plane[-self.dims.ndisplay :] = 0
+            mapped_clipping_plane = inv_transform(clipping_plane)
+            if not np.allclose(
+                mapped_clipping_plane[-self.dims.ndisplay :], 0
+            ):
+                warnings.warn(
+                    'Non-orthogonal slicing is being requested, but'
+                    ' is not fully supported.'
```

Comment:
"fully"? Also, what actually happens in this case? The warning should specify this, e.g. "displaying data without applying out-of-slice rotation component."

Reply:
exactly, i have added